lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -47,7 +47,11 @@ def get_stat
       raw.each_with_index do |line, index|
         mount_path = path[index]
         values = normalize_values(line.split)
-        @stat[@paths_hash.key(mount_path)] = Hash[titles.zip(values)].merge({:path => mount_path, :size => SIZE.key(size)})
+        if RUBY_VERSION.start_with? "1.8"
```

Comment:
If you are looking up a hash key based on its value, you may as well use Hash#find (or #select if you expect more than one match) -- it's going to iterate over all of the values in the map anyway, but it's going to be platform-independent.

I would suggest, however, looking into how ```@paths_hash`` is used -- looks like you can replace it with a Map of paths to keys (:pulp_dir, etc), in which case you can use a normal hash lookup.

Similarly, it looks to me that a reversed lookup in SIZE map can be replaced with a value of ```opts[:size] || :kilobyte``` expression stored in an instance variable...